src/quick_sort.py

```diff
@@ -0,0 +1,90 @@
+"""Module that contains a function with an implementation of the quick sort."""
+
+
+def quick_sort(lst):
+    """Function that implements the quick sort."""
+    if not isinstance(lst, list):
+        raise ValueError('Input must be a list.')
+    if not all(isinstance(item, (float, int, str)) for item in lst):
+        raise ValueError('Items in list must be integers, strings, or floats.')
+    if len(lst) <= 1:
+        return
+    pivot = lst[0]
+    right_list = []
+    left_list = []
+    for val in lst[1:]:
+        if val > pivot:
+            right_list.append(val)
+        else:
+            left_list.append(val)
+    quick_sort(left_list)
+    quick_sort(right_list)
+    return left_list + [pivot] + right_list
+
+
+if __name__ == '__main__':  # pragma no cover
```

Comment:
Note that if you're doing a comparison of best and worst-case times, it might be best to check against the same length of numbers each time.